pkg/objectfile/pool.go

```diff
@@ -139,6 +146,15 @@ func (p *Pool) Get(buildID string) (*ObjectFile, error) {
 // The returned reference should be released after use.
 // The file will be closed when the reference is released.
 func (p *Pool) Open(path string) (*ObjectFile, error) {
+	if val, ok := p.buildIDCache.GetIfPresent(path); ok {
+		buildID, ok := val.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected type in cache: %T", val)
+		}
+
+		return p.get(buildID)
```

Comment:
Im equally unsure about this. Selecting things based on an opaque ID that’s in user’s control doesn’t seem like a good idea unless it’s exactly the same file.

Reply:
Yes I’m happy with iterating on this. In that case I’d not just hash the .text but the whole file instead.